### PR TITLE
Test Facebook Native Ads

### PR DESCRIPTION
**Pull request by regiskuckaertz**
Will insert a Facebook HTML ad on mobile devices before the first container when the adtest cookie is set to `fbnative`.

Works locally, now needs to go online for stakeholders to see.

picture 1

_It's ugly but it's not the point_

@jbreckmckye @uplne @JonNorman @rich-nguyen @Calanthe

### CODE REVIEW

File: static/src/stylesheets/module/commercial/_facebook.scss [hunk 1]
```diff
@@ -0,0 +1,32 @@
+.fb-ad-slot {
```
Comment (jbreckmckye) [inline static/src/stylesheets/module/commercial/_facebook.scss]: I guess all these styles are temporary?

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+            referenceNode: '#top > .l-side-margins > :first-child',
+            where: 'beforebegin',
+            placementId: '978824118882656_978826235549111',
+            adId: 'ad_root'
+        }
+    ];
+    var facebookTpl;
+
+    return {
+        load: init
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: I _think_ I can see why we've named this method this way - `load` has more meaning for consumers of the module; `init` has more meaning inside the module - but couldn't we expose just a singular `loadFacebook` function?

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    'common/utils/cookies',
+    'common/utils/template',
+    'text!common/views/commercial/facebook.html'
+], function(fastdom, detect, cookies, template, facebookStr) {
+    var scriptId = 'facebook-jssdk';
+    var scriptSrc = '//connect.facebook.net/en_US/sdk/xfbml.ad.js#xfbml=1&version=v2.5';
+    var appNetworkId = '978824118882656';
+    var adUnits = [
+        {
+            referenceNode: '#top > .l-side-margins > :first-child',
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: Looking at this I don't think I get any sense that this is the first container

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            where: 'beforebegin',
+            placementId: '978824118882656_978826235549111',
+            adId: 'ad_root'
+        }
+    ];
+    var facebookTpl;
+
+    return {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: I'd probably just shove this to the bottom.

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+        // we need the SCRIPT element to have an ID so we won't be using curl.js
+        // this time, sorry guys
+        insertFbScript();
+    }
+
+    function insertAdUnits() {
+        facebookTpl = template(facebookStr);
+        adUnits.forEach(function (adUnit) {
+            adUnit.referenceNode = document.querySelector(adUnit.referenceNode);
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: I'm not sure if I'm keen on this kind of object mutation...

File: static/src/stylesheets/module/commercial/_facebook.scss (hunk 1 above)
Comment (regiskuckaertz) [inline static/src/stylesheets/module/commercial/_facebook.scss]: Yep

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js (hunk 3 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: This is temporary too, I didn't want to think about it too much 😅

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+                }
+            );
+        };
+
+        insertAdUnits();
+
+        // we need the SCRIPT element to have an ID so we won't be using curl.js
+        // this time, sorry guys
+        insertFbScript();
+    }
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: Maybe `init()` should take responsibility for freeing up memory.

File: static/src/javascripts/projects/common/views/commercial/facebook.html [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<div class="fb-ad" data-placementid="<%=placementId%>" data-format="native" data-nativeadid="<%=adId%>" data-testmode="false"></div>
+<div class="fb-ad-slot" id="<%=adId%>">
+    <a class="fbAdLink">
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/views/commercial/facebook.html]: What does this class do?

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js (hunk 3 above)
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: Fair enough!

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js (hunk 2 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: This is to mimic the signature of other 3rd parties (in `third-party-tags`), although I'm not sure yet this module will remain there for long... we need to address questions such as which is the fallback of which: DFP or Facebook?

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js (hunk 5 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: Oh, I always think `forEach` implies some kind of mutation.

File: static/src/javascripts/projects/common/views/commercial/facebook.html (hunk 7 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/views/commercial/facebook.html]: It is used by the FB lib, I can't change these unfortunately.

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js (hunk 6 above)
Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: Bah, this is a small thing I can improve later, I don't want to chain promises for now. I'll probably remove it at some point.

File: static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js (hunk 2 above)
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/third-party-tags/facebook.js]: Yeah, could be some interesting questions to resolve. I think this is fine as is.

**Comment by jbreckmckye** [comment]
Progress marches on!

👍

**Pull request merged**